### PR TITLE
Dockerise mock sds

### PR DESCRIPTION
**Pull request by katie-gardner**
### What is the context of this PR?
This PR spins up the mock SDS service alongside runners other supporting services to enable easy deployment and functional testing with supplementary data.

### How to review
Check the requirements in the card are satisfied and that the commands in the readme files work as expected. Check the previously skipped tests are no longer skipped and check runner and launcher can talk to the mock SDS service after it has been launched with the extended `make dev-compose-up` command

### Checklist
* [ ] New static content marked up for translation
* [ ] Newly defined schema content included in eq-translations repo
* [ ] Functional tests using supplementary data are running and passing

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -115,6 +115,12 @@ make dev-compose-up-linux
 docker run -e SURVEY_RUNNER_SCHEMA_URL=http://docker.for.mac.host.internal:5000 -it -p 8000:8000 onsdigital/eq-questionnaire-launcher:latest
```
Comment (MebinAbraham) [inline README.md:115]: We need to set SDS API URL the same way here?.

File: .development.env [hunk 2]
```diff
@@ -27,5 +27,5 @@ CDN_ASSETS_PATH=/design-system
 ADDRESS_LOOKUP_API_URL=https://whitelodge-ai-api.census-gcp.onsdigital.uk
 COOKIE_SETTINGS_URL=#
 EQ_SUBMISSION_CONFIRMATION_BACKEND=log
-SDS_API_BASE_URL=http://localhost:5003
-OIDC_TOKEN_BACKEND=local
\ No newline at end of file
+SDS_API_BASE_URL=http://sds:5003
```
Comment (MebinAbraham) [inline .development.env:30]: The issue with updating this is that, this file can now not be used when running runner locally. This expects runner to be running in docker.

I suggest reverting this but in docker-compose.yml set `SDS_API_BASE_URL: "http://sds:5003"` inline which will take precedence. Then, if you use this file for local runner or when running in docker, both cases work.

File: .functional-tests.env [hunk 3]
```diff
@@ -28,5 +28,5 @@ ADDRESS_LOOKUP_API_URL=https://whitelodge-ai-api.census-gcp.onsdigital.uk
 COOKIE_SETTINGS_URL=#
 EQ_SUBMISSION_CONFIRMATION_BACKEND=log
 VIEW_SUBMITTED_RESPONSE_EXPIRATION_IN_SECONDS=35
-SDS_API_BASE_URL=http://localhost:5003
+SDS_API_BASE_URL=http://sds:5003
```
Comment (MebinAbraham) [inline .functional-tests.env:31]: Although this is mainly only used by action, the comment above is appropriate here also.

File: .development.env (hunk 2 above)
Comment (katie-gardner) [reply .development.env]: Ahh makes sense, have made that change and switched the dev compose files to use the same approach as the runner url

File: README.md (hunk 1 above)
Comment (katie-gardner) [reply README.md]: Added 👍

File: .development.env (hunk 2 above)
Comment (MebinAbraham) [reply .development.env]: Doesn't look like `docker-compose-yml` was updated.

File: docker-compose.yml [hunk 4]
```diff
@@ -2,47 +2,56 @@ version: '3'
...
+            - eq-env
+
+    eq-questionnaire-runner:
+        image: onsdigital/eq-questionnaire-runner:latest
+        build: ./
+        env_file:
+            - ${RUNNER_ENV_FILE}
+        environment:
+            DATASTORE_EMULATOR_HOST: datastore:8432
+            EQ_REDIS_HOST: "redis"
```
Comment (MebinAbraham) [inline docker-compose.yml:22]: https://github.com/ONSdigital/eq-questionnaire-runner/pull/1187#discussion_r1297461180 needs to be done here as well.

Comment (katie-gardner) [reply docker-compose.yml:22]: Thanks 👍 added it here

File: .development.env (hunk 2 above)
Comment (katie-gardner) [reply .development.env]: oops - updated as per other comment

**Review by MebinAbraham — APPROVED** [review]

**Review by berroar — APPROVED** [review]

**Review by petechd — APPROVED** [review]

**Pull request merged**